### PR TITLE
Add Margin support

### LINKED ISSUE #52: [Feature Request] Add margin Trading
This could be slightly complicated, so this is more of an idea what could be implemented. and a place to store information

information regarding margin trading in germany:

> Kapitalerträge aus Margin Trading
> 
> Schließlich werden in dieser Kategorie alle Gewinne und Verluste aus Margin Trading abgebildet, also solche An- und Verkäufe die „auf Marge", das heißt mit einem Hebel durchgeführt wurden. 
> Margingewinne, -verluste und -gebühren werden in dieser Übersicht aufsummiert dargestellt, weil sie nicht dem persönlichen Einkommensteuersatz unterliegen. Sie werden pauschal mit einem Steuersatz von 25 Prozent der Kapitalertragsteuer unterworfen.
> 

https://www.accointing.com/de/crypto-tax-guide/de/wie-versteuere-ich-verluste-aus-dem-cryptohandel

> Berücksichtigung von Verlusten aus Margin Trading nur begrenzt möglich
> 
> Verluste aus Kapitalerträgen, wie z.B. durch Margin Trading mit Kryptowährungen auf BitMex, können auch berücksichtigt werden. 
> Allerdings ist hier der Verlustabzug ab dem 01.01.2020 aufgrund der neuen „Derivatesteuer“ nur noch sehr begrenzt möglich. 
> Eine vertikale Verrechnung von Verlusten aus der Anlage KAP und der Anlage SO war auch schon vor der Gesetzesänderung aufgrund der Verlustabzugsbeschränkungen ausgeschlossen.

https://winheller.com/blog/verluste-bitcoin-kryptowaehrungen-steuererklaerung/

haven't found more till now

### PR DESCRIPTION
**Pull request by scientes**
This needs to be tested further

closes #52 

and i'm not sure if the gain/loss calculation is 100% correct because i'm not sure if we'd need to make seperate balance counters for margin and spot, but seperating those would be a major hassle

### CODE REVIEW

**Comment by scientes** [comment]
due it now seperating losses form gains this should be fine because now tist is seperated from spot gains

**Comment by provinzio** [comment]
I am not that familiar with the taxation of margin trades. Bit aren't these CFD? And CFD are taxed with "Kapitalertragssteuer" (25%). 

Plead add an description/explanation on margin trade to the readme and cite a trustworthy source.

**Comment by scientes** [comment]
Accointing (a german online service for crypto taxation) lists profits and losses from margin trading in a seperate colummn. 

yes "Kapitalertragssteuer" (25%) apply here but losses from margin trading are complicated, so to help that both wins and losses are displayed seperately:

> Verluste aus Kapitalerträgen, wie z.B. durch Margin Trading mit Kryptowährungen auf BitMex, können auch berücksichtigt werden. Allerdings ist hier der Verlustabzug ab dem 01.01.2020 aufgrund der neuen „Derivatesteuer“ nur noch sehr begrenzt möglich. Eine vertikale Verrechnung von Verlusten aus der Anlage KAP und der Anlage SO war auch schon vor der Gesetzesänderung aufgrund der Verlustabzugsbeschränkungen ausgeschlossen.

now that you say that the duplication does not make sense anymore, because at first i tried to seperate margin and spot but that doesn't work that well, but on the export the MarginBuy and sell does help differntiate a little especially for debugging

**Comment by scientes** [comment]
Should We maybe name it Margin: Kapitalertragssteuer?

**Comment by provinzio** [comment]
> Accointing (a german online service for crypto taxation) lists profits and losses from margin trading in a seperate colummn.
> https://accointing.com/de/crypto-tax-guide/de/was-wird-fur-das-finanzamt-benotigt
> 
> yes "Kapitalertragssteuer" (25%) apply here but losses from margin trading are complicated, so to help that both wins and losses are displayed seperately:
> 
> > Verluste aus Kapitalerträgen, wie z.B. durch Margin Trading mit Kryptowährungen auf BitMex, können auch berücksichtigt werden. Allerdings ist hier der Verlustabzug ab dem 01.01.2020 aufgrund der neuen „Derivatesteuer“ nur noch sehr begrenzt möglich. Eine vertikale Verrechnung von Verlusten aus der Anlage KAP und der Anlage SO war auch schon vor der Gesetzesänderung aufgrund der Verlustabzugsbeschränkungen ausgeschlossen.
> 
> https://winheller.com/blog/verluste-bitcoin-kryptowaehrungen-steuererklaerung/
> 
> now that you say that the duplication does not make sense anymore, because at first i tried to seperate margin and spot but that doesn't work that well, but on the export the MarginBuy and sell does help differntiate a little especially for debugging

Please check other sources too. Crypto taxation is quite a gray area.

Summarize your findings in the readme.

**Comment by Griffsano** [comment]
Hey,

Winheller published an article in October that goes more into detail: https://winheller.com/blog/besteuerung-future-margin-trading/

> Parallel dazu lassen sich die Überlegungen auf das Margin Trading übertragen, weshalb Gewinne aus Margin Trades immer nur dann unter Kapitaleinkünfte (§ 20 EStG) fallen, wenn keine Lieferung einer Kryptowährung, sondern ein Differenzausgleich durchgeführt wird. Kommt es hingegen zu einer Lieferung einer Kryptowährung, liegt ein privates Veräußerungsgeschäft gemäß § 23 Abs. 1 Satz 1 Nr. 2 EStG vor.

For my understanding that means if a margin position is closed, the gain/loss falls under income from capital (25%), and if a position is settled (you buy/sell the coin at the end wirh the starting price), you can treat it just as spot trades.

Additionally, according to the article, fees for futures may not be subtracted from the capital income, as they fall under "Werbungskosten" and are already covered by the standard deduction / "Pauschbetrag", and losses may only be considered up to 20k with the new regulation. However, I'm not sure if this applies to closed positions as well.

**Comment by provinzio** [comment]
> Additionally, according to the article, fees for futures may not be subtracted from the capital income, as they fall under "Werbungskosten" and are already covered by the standard deduction / "Pauschbetrag", and losses may only be considered up to 20k with the new regulation. In my opinion, this applies then to closed positions.

We should still calculate the "Werbungskosten" in case the user has more expenses than the "Pauschbetrag".

**Comment by Griffsano** [comment]
For the exchanges that I've seen, there's always a distinction between futures and margin trading. Compared to futures (with or without leverage), margin trading is spot trading with leverage. This is also what [Kraken](https://www.kraken.com/features/margin-trading) states:
> Margin Trading allows you to make spot purchases and sales of cryptocurrencies, on the Kraken exchange, using funds that may exceed the balance of your account.

I'm pretty certain that (classical) "closed" margin trades (which lead to the payment of the difference in value) are per definition CFDs/"Differenzkontrakte" and therefore "Termingeschäfte" (similar to classical futures). This also agrees with multiple articles / forum threads.

For "settled" margin trades (that lead to the delivery of the asset for the base price, not the difference in value), it could be different (I also found this opinion in a couple of threads).
Argument 1 against "Termingeschäft": [Wikipedia](https://de.wikipedia.org/wiki/Termingesch%C3%A4ft) states that there needs to be a pre-defined point in the future:
> Termingeschäfte (auch Zeitgeschäfte; englisch forward oder englisch future) sind Finanzkontrakte, bei denen sich die Vertragsparteien verpflichten, die gegenseitige Erfüllung (Zahlung und Lieferung) über mehr als zwei Handelstage hinaus auf einen vereinbarten Zeitpunkt zu verschieben.

For the exchanges I know, margin trades can be settled at any non-predefined point in time (for Kraken up to a duration of max. 365 days). This argument would also apply to "closed" margin trades, if they wouldn't already be CFDs.
Argument 2: [Winheller](https://winheller.com/blog/besteuerung-future-margin-trading) differentiates between the delivery of the asset or the payment of the difference in value (they apply the same argumentation to futures btw):
> Parallel dazu lassen sich die Überlegungen auf das Margin Trading übertragen, weshalb Gewinne aus Margin Trades immer nur dann unter Kapitaleinkünfte (§ 20 EStG) fallen, wenn keine Lieferung einer Kryptowährung, sondern ein Differenzausgleich durchgeführt wird. Kommt es hingegen zu einer Lieferung einer Kryptowährung, liegt ein privates Veräußerungsgeschäft gemäß § 23 Abs. 1 Satz 1 Nr. 2 EStG vor.

Argument 3: [Accointing](https://hub.accointing.com/crypto-tax-regulations/germany/tax-break-germany-derivate-und-futures-winheller) gives this example scenario:
> Person A schließt mit Person B einen Vertrag, der A das Recht einräumt, in Zukunft einen BTC von B zu erhalten, der momentan 35.000 Euro wert ist. Wird der Kontrakt fällig und A erhält von B den BTC, liegt ein Fall von § 23 Abs. 1 Satz 1 Nr. 2 EStG vor. Das heißt für B, dass er die Veräußerung des BTC mit seinem persönlichen Einkommensteuersatz versteuern muss. Für A bedeutet das hingegen, dass eine Anschaffung vorliegt und damit die Jahresfrist gilt.

The example scenario on [Accointing](https://hub.accointing.com/crypto-tax-regulations/germany/tax-break-germany-derivate-und-futures-winheller) continues as follows:
> Treffen A und B hingegen im oben geschilderten Fall die Vereinbarung, dass A am Ende des Vertrages die Wahl hat, ob er einen BTC bekommt oder alternativ den Gegenwert der Differenz zum aktuellen Kurs, dann liegt ein Termingeschäft gem. § 20 Abs. 2 Satz 1 Nr. 3 EStG vor.

At least for Kraken, the user has the option of "settling" (delivery of asset) or "closing" (difference in value) a margin trade, the decision is open until the end.

In summary, I arrive at this understanding:
- If the user has the option of "closing" or "settling", it's a "Termingeschäft" -> Valid for all Kraken margin trades, not sure how it is handled on other exchanges. However, this is only supported by the [Accointing](https://hub.accointing.com/crypto-tax-regulations/germany/tax-break-germany-derivate-und-futures-winheller) example scenario.
- Else, if the trade is "closed", it's a "Termingeschäft"
- Otherwise, the trade is "settled" (the asset is delivered in the end), and it's a "privates Veräußerungsgeschäft"

**Comment by provinzio** [comment]
Ah ok. Sounds good. Thanks for the detailed explanation.

**Comment by provinzio** [comment]
Stale. Further worked on in #120
